ksonnet-gen/ksonnet/emit.go

```diff
@@ -85,60 +159,83 @@ func (root *root) emit(m *indentWriter) {
 	m.indent()
 
 	// Emit in sorted order so that we can diff the output.
+	groups := root.groups.toSortedSlice()
+	var groupNames []string
+	for _, g := range groups {
+		groupNames = append(groupNames, string(g.name))
+	}
+
 	for _, group := range root.groups.toSortedSlice() {
```

Comment:
We're calling `root.groups.toSortedSlice()` twice (once here and once on 162). This should be deterministic, so it seems like we might as well just re-use instead of sorting twice?